File: credit/credit.py
from cs50 import get_int
from cs50 import get_string


cc = get_string("Number: ")
sum = 0
sum2 = 0
# last digit
for i in range(len(cc)-1, -1, -2):
    sum = sum + int(cc[i])

digit1 = 0
digit2 = 0
# 2nd to the last digit
for i in range(len(cc)-2, -1, -2):
    digit1 = int(cc[i]) * 2 // 10
    digit2 = int(cc[i]) * 2 % 10
    # add each digits
    sum2 = sum2 + digit1 + digit2
total_sum = sum + sum2
firsttwo = int(cc[0]) * 10 + int(cc[1])

# checks for valid CC
if(total_sum % 10 != 0):
    print("INVALID")
# checks for AMEX
elif((firsttwo == 34 or firsttwo == 37) and len(cc) == 15):
    print("AMEX")
# checks for MC
elif((firsttwo >= 51 and firsttwo <= 55) and len(cc) == 16):
    print("MASTERCARD")
# checks for Visa
elif(int(cc[0]) == 4 and (len(cc) == 13 or len(cc) == 16)):
    print("VISA")
# else return invalid
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //ask user to input the CC number
    long long credit = get_long_long("Enter the number:\n");
    
    int lastdigit = 0, count = 0, dble_odd_number = 0, odd_number = 0, even_number = 0, sum = 0, seconddigit = 0, firsttwo = 0;
    
    //loop that removes a digit until there are no more numbers left
    while (credit > 0)
    {
    //saves 2nd digit before last digit gets replaced
        seconddigit = lastdigit;    
        lastdigit = credit % 10;
    //adds odd & even numbers
        if (count % 2 == 0)
        {
            even_number +=  lastdigit;
        }
        else
        {
            dble_odd_number = lastdigit * 2;
            odd_number += (dble_odd_number / 10 + dble_odd_number % 10) ;
        }
    credit /= 10;
    count++;
    }
    //checking if the number is valid
    sum = even_number + odd_number;
    bool valid = sum % 10 == 0;
    firsttwo = lastdigit * 10 + seconddigit;
    if ((firsttwo == 34 || firsttwo == 37) && count == 15 && valid)
    {
        printf("AMEX\n");
    }
    else if (firsttwo >= 51 && firsttwo <= 55 && count == 16 && valid)
    {
        printf("MASTERCARD\n");
    }
    else if (lastdigit == 4 && count >= 13 && count <= 16 && valid)
    {
        printf("VISA\n");
    }
    else
        printf("INVALID\n");        
    }